light/verifier.go

```diff
@@ -177,6 +178,14 @@ func ValidateTrustLevel(lvl tmmath.Fraction) error {
 	return nil
 }
 
+func ValidateNumSamples(numSamples uint32) error {
+	maxShares := consts.MaxSquareSize * consts.MaxSquareSize
+	if numSamples == 0 && numSamples < uint32(maxShares) {
```

Comment:
`maxShares` should never be `0`, so the second condition is trivially true if the former is true. Should remove (and ideally place these assumptions in a separate file with static assertions like https://github.com/bitcoin-core/secp256k1/blob/3dc8c072b6d84845820c1483a2ee21094fb87cc3/src/assumptions.h).

Also, this should be `numSamples > uint32(maxShares)`.

Reply:
Great find. Removed the 2nd condition as the code will calculate the actual number of samples as:
```go
numSamples := min(c.numSamples, uint32(numRows*numRows))
````

> ideally place these assumptions in a separate file with static assertions like https://github.com/bitcoin-core/secp256k1/blob/3dc8c072b6d84845820c1483a2ee21094fb87cc3/src/assumptions.h).

What does static mean in this context? The other checks like this are in this file. I kept the new one here for consistency with the existing code.

Reply:
Static as in your program doesn't even compile without an error if these assumptions are violated. I don't know if Go has such a thing, but something like a small module that runs when the program starts that panics if certain fundamental assumptions are violated (e.g. the platform not being 64 bits, or having set a parameter to a value that was assumed impossible like setting the max size to `0`). Either way it's orthogonal to this PR.

Reply:
Yeah, that does not work here as the inputs that are checked are user input (not known during compile time).

Reply:
>  I don't know if Go has such a thing, but something like a small module that runs when the program starts

https://golang.org/doc/effective_go#init

but this is also during run time, not compile time.

Reply:
I meant that the max size parameter is not set to 0. That's a compile-time constant, and parts of the code assume it's > 0.

Crashing the program before it even does anything, deterministically, isn't great...but it's better than crashing in the middle randomly based on user input.